src/dmd/argtypes.d

```diff
@@ -434,8 +502,10 @@ TypeTuple toArgTypes(Type t)
                      * Cannot find any documentation on it.
                      */
 
-                    VarDeclaration f = t.sym.fields[0];
-                    TypeTuple tup = toArgTypes(f.type);
+                    uint foffset;
+                    uint falignsize;
+                    Type ftype = getFieldInfo(0, foffset, falignsize);
+                    TypeTuple tup = toArgTypes(ftype);
```

Comment:
I wonder who originally added this. It seems a bit odd to have a FreeBSD special case here.

Reply:
I think it was me. Nevertheless, it is for real.

Reply:
https://github.com/dlang/dmd/pull/7230

Found it and commented there. Very certain that this should be turned on for other non-Windows x86 platforms too.  I've been looking at the gcc x86 backend recently to implement what this module effectively does in gcc itself.